File: code-interpreter/src/main/kotlin/cc/unitmesh/code/messaging/Message.kt
package cc.unitmesh.code.messaging

import kotlinx.serialization.Serializable

@Serializable
data class Message(
    var id: Int = -1,
    var resultValue: String,
    var displayValue: String,
    var className: String = "",
    var msgType: MessageType = MessageType.NONE,
    var content: MessageContent? = null,
)

File: choc-ui/components/interpreter/message.ts
export type InterpreterMessage = {
    id: number
    resultValue: string
    displayValue: string
    className: string
    msgType: MsgType
    content: MessageContent
}

export type MsgType = "none" | "error" | "html" | "running"

export type MessageContent = {
    type: string
    // when type is "cc.unitmesh.code.messaging.HtmlContent"
    html?: string
}
